packages/teams-js/test/internal/communication.spec.ts

```diff
@@ -414,47 +412,35 @@ describe('Testing communication', () => {
 
     it('unresolved message callbacks should not be triggered after communication has been uninitialized', () => {
       app._initialize(utils.mockWindow);
-      GlobalVars.isFramelessWindow = false;
-      communication.Communication.parentWindow = utils.mockWindow.parent;
-      communication.Communication.parentOrigin = utils.validOrigin;
-      communication.Communication.currentWindow = utils.mockWindow;
+      communication.initializeCommunication(undefined);
       let callbackWasCalled = false;
+      const initializeMessage = utils.findInitializeMessageOrThrow();
+      utils.respondToMessage(initializeMessage);
 
       communication.sendMessageToParent('testAction', () => {
         callbackWasCalled = true;
       });
+      const tempProcessMessage = utils.processMessage;
       communication.uninitializeCommunication();
-      const secondMessage = utils.findMessageByFunc('testAction');
-      communication.processMessage({
-        originalEvent: { originalEvent: {} as DOMMessageEvent, func: '' },
-        func: '',
-        data: { id: secondMessage?.id, args: [] },
-        source: communication.Communication.parentWindow,
-        origin: utils.mockWindow.location.origin,
-      });
+      utils.processMessage = tempProcessMessage;
+
+      utils.respondToMessage({ id: 1, func: 'testAction' }, false);
 
       expect(callbackWasCalled).toBeFalsy();
     });
 
     it('unresolved message promises should not be triggered after communication has been uninitialized', async () => {
       expect.assertions(1);
       app._initialize(utils.mockWindow);
-      GlobalVars.isFramelessWindow = false;
-      communication.Communication.parentWindow = utils.mockWindow.parent;
-      communication.Communication.parentOrigin = utils.validOrigin;
-      communication.Communication.currentWindow = utils.mockWindow;
+      communication.initializeCommunication(undefined);
 
       const messageParent = communication.sendMessageToParentAsync('testAction');
 
+      const tempProcessMessage = utils.processMessage;
       communication.uninitializeCommunication();
-      const secondMessage = utils.findMessageByFunc('testAction');
-      communication.processMessage({
-        originalEvent: { originalEvent: {} as DOMMessageEvent, func: '' },
-        func: '',
-        data: { id: secondMessage?.id, args: [] },
-        source: communication.Communication.parentWindow,
-        origin: utils.mockWindow.location.origin,
-      });
+      utils.processMessage = tempProcessMessage;
+
+      utils.respondToMessage({ id: 1, func: 'testAction' }, false);
 
       messageParent.then(() => expect(true).toBeTruthy());
```

Comment:
If I am reading/understanding this test correctly, we don't expect that the `then` block will ever be executed, and only the expect below it will be executed? And if the `then` block is executed then the test will fail because jest saw two assertions instead of one. Is that correct?

Does it seem like it would be more straightforward to remove the assertion count expectation and instead just make the assertion inside the `then` block be an unconditional failure?

I don't know if some async aspect of this wouldn't work then... it was just hard for me to see what appeared to be a "true" assertion inside the then block and think "wait, why will this fail?"

Reply:
Perhaps you could just call `fail()` inside the `then` block?

Reply:
Very fair. I can't think of a reason why what you're asking for would break the async-ness of the test, and it's definitely easier to read.